### PR TITLE
completed question-2

### PR DESCRIPTION
**Pull request by sonalikari**
- Created account token during login using a random number generate e.g md5. Save this in a collection “access_token”.
- Created a new route /user/address of method POST
- Created a new route /user/get/:id of type GET

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        res.status(400).json({ error: "Error occurred! Try Again" });
+    }
+};
+
+exports.loginUser = async (req, res) => {
+    try {
+        const { username, password } = req.body;
+        const user = await User.findOne({ username });
+
+        if (!user || !(await bcrypt.compare(password, user.password))) {
```
Comment (anmolint) [inline controllers/userController.js]: This can be done in validator

File: models/User.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    phone: { type: String, required: true }
+});
+
+const userSchema = new mongoose.Schema({
+    username: { type: String, required: true, unique: true },
+    password: { type: String, required: true },
+    email: { type: String, required: true, unique: true },
+    firstname: { type: String, required: true },
+    lastname: { type: String, required: true },
+    addresses: [addressSchema]
```
Comment (anmolint) [inline models/User.js]: use refrencing for now embedding is not required

File: routes/userRoutes.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// Routes
+router.post('/register', validationMiddleware.validateRegistration, userController.registerUser);
+router.post('/login', validationMiddleware.validateLogin, userController.loginUser);
+
+router.get('/get', authMiddleware.validateAccessToken, userController.getUserData);
+router.put('/delete', authMiddleware.validateAccessToken, userController.deleteUserData);
+
+router.get('/list/:page', userController.getUserList);
+
+router.post('/address', authMiddleware.validateAccessToken, userController.addUserAddress);
```
Comment (anmolint) [inline routes/userRoutes.js]: why no validator for addng address?

File: routes/userRoutes.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const userController = require('../controllers/userController');
+const validationMiddleware = require('../middleware/validationMiddleware');
+const authMiddleware = require('../middleware/authMiddleware');
+
+// Routes
+router.post('/register', validationMiddleware.validateRegistration, userController.registerUser);
+router.post('/login', validationMiddleware.validateLogin, userController.loginUser);
+
+router.get('/get', authMiddleware.validateAccessToken, userController.getUserData);
+router.put('/delete', authMiddleware.validateAccessToken, userController.deleteUserData);
```
Comment (anmolint) [inline routes/userRoutes.js]: can we use a different verb here?

Comment (sonalikari) [reply routes/userRoutes.js]: In place of put?

File: middleware/validationMiddleware.js [hunk 5]
```diff
@@ -21,7 +23,22 @@ exports.validateRegistration = [
     }
 ];
 
-exports.validateLogin=[
+exports.validateLogin = [
     body('username').notEmpty().withMessage('Username is required'),
-    body('password').notEmpty().withMessage('Password is required')
+    body('password').notEmpty().withMessage('Password is required').custom(async (password, { req }) => {
```
Comment (anmolint) [inline middleware/validationMiddleware.js:22]: needed a check for password === confirmpassword why we moved the lofin logic here?

Comment (anmolint) [reply middleware/validationMiddleware.js:22]: for register

Comment (sonalikari) [reply middleware/validationMiddleware.js:22]: Already mentioned validation for password and confirm password in validateRegistration(line no. 7 to 12).

Comment (anmolint) [reply middleware/validationMiddleware.js:22]: ok but the password match logic should be inside controller move itinto controller ,merge it then move to next question

**Pull request merged**